src/global/ObjectTest.java

```diff
@@ -0,0 +1,485 @@
+            validateMethodTestSentence(methodName, methodTestSentence);
+            Matcher matcher = getMatches(output, processRegexForPrintlnOutput(combineRegex(methodTestSentence)));
+            if (incorrectMethodStructureErrorMessage.equals(""))
+                incorrectMethodStructureErrorMessage = String.join("", "Your ", methodName, " method output did not follow the correct structure/syntax.");
+            _assertTrue(matcher.find(), incorrectMethodStructureErrorMessage, parseTestInformation(output, methodTestSentence, incorrectMethodStructureErrorMessage));
+            // Ensures that the output matches the pattern exactly
+            assertEquals(output.substring(matcher.start(), matcher.end()), output, incorrectMethodStructureErrorMessage);
+
+            // check all clauses valid
+            int matchGroupNum = 1;  // match group numbers are 1-indexed
+            for (Clause clause : methodTestSentence) {
+                // TODO: student-facing message can include details + feedback should be dynamic to the clause's error
+                _assertTrue(clause.validate(matcher.group(matchGroupNum)), incorrectMethodStructureErrorMessage, "Invalid Clause output at index " + matchGroupNum + " :" + clause.getInvalidMessage());
+                matchGroupNum++;
+            }
+        }
+    }
+
+    private void validateMethodTestSentence(String methodName, Clause[] methodTestSentence) throws InvalidClauseException {
+        /*
+         This method makes sure that the clause structure is defined correctly, and the test writer does not use any
+         features which are not allowed in methods. Parts of this method are adapted from the setRegexSentence in
+         BaseTest.
+         */
+
+        //By using a set we can ensure that there are no duplicates in the regexSentence
+        Set<String> namesSet = new HashSet<>();
+
+        int i = 1;
+        for (Clause clause : methodTestSentence) {
+            if (clause instanceof RandomClause || clause instanceof PlaceHolder)
+                _fail("There is an issue with the test definition. Please contact an administrator.",
+                        "Error with test definition: The method clause system does not support random clauses or the placeHolder clause.");
+            if (clause.getName() != null) {
+                if (namesSet.contains(clause.getName())) {
+                    _fail("There is an issue with the test definition. Please contact an administrator.",
+                            "The name " + clause.getName() + " is already in use. Make sure all names are unique");
+                }
+                namesSet.add(clause.getName());
+            }
+            try {
+                clause.validateParams();
+            } catch (InvalidClauseException e) {
+                throw new InvalidClauseException("The parameter(s) for the " + methodName + " method clause at index " + i + " are invalid. " + e.getMessage());
```

Comment:
What will happen when we throw an error without `_fail`? What does the user see?

Reply:
I think they will see this, but this should be changed to fail.

Reply:
This is actually a developer error if the clause is not defined correctly, so the student should never see this error. However, I will change it to use _fail anyways.

Reply:
Unless there is an error with our test, but they should be tested as they are put on the website to make sure this doesn't happen.